src/ts/components/NavigatorComponent.ts

```diff
@@ -0,0 +1,103 @@
+import ProductInputComponent from './ProductManageComponent/ProductInputComponent';
+import ProductStateComponent from './ProductManageComponent/ProductsStateComponent';
+import CoinInputComponent from './CoinManageComponent/CoinInputComponent';
+
+import VendingMachineProductManager from '../domains/VendingMachineProductManager';
+import VendingMachineCoinManager from '../domains/VendingMachineCoinManager';
+import CoinsStateComponent from './CoinManageComponent/CoinsStateComponent';
+
+import { $, on } from '../dom';
+import { ROUTES } from '../constants';
+
+export default class NavigatorComponent {
+  private $productInfoSection: HTMLElement = $('.product-info-section');
+  private $chargeCoinSection: HTMLElement = $('.charge-coin-section');
+  private $navProductButton = $('.nav__product-button') as HTMLButtonElement;
+  private $navChargeButton = $('.nav__charge-button') as HTMLButtonElement;
+  private $coinInput = $(
+    '.charge-form-section__coin-input'
+  ) as HTMLInputElement;
+  private $productInput = $(
+    '.product-info-form__product-input'
+  ) as HTMLInputElement;
+
+  private vendingMachineProductManager = new VendingMachineProductManager();
+  private vendingMachineCoinManager = new VendingMachineCoinManager();
+
+  constructor() {
+    new ProductStateComponent(this.vendingMachineProductManager);
+    new ProductInputComponent(this.vendingMachineProductManager);
+    new CoinInputComponent(this.vendingMachineCoinManager);
+    new CoinsStateComponent();
+
+    on(this.$navProductButton, 'click', this.onClickNavProductButton);
+    on(this.$navChargeButton, 'click', this.onClickNavChargeButton);
+    on(window, 'popstate', this.onPopstateRoute);
+    this.routeURLVisit(window.location.pathname);
+  }
+
+  private routeURLVisit(pathname: string): void {
+    if (
+      !Object.values(ROUTES).some((route) => route === window.location.pathname)
+    ) {
+      window.history.replaceState(null, null, '/');
+
+      return;
+    }
+
+    if (pathname === ROUTES.COINS) {
+      this.renderCoinComponent();
+      window.history.pushState({}, '', ROUTES.COINS);
+
+      return;
+    }
+
+    if (pathname === ROUTES.PRODUCTS) {
+      this.renderProductComponent();
+      window.history.pushState({}, '', ROUTES.PRODUCTS);
+
+      return;
+    }
+  }
+
+  private onPopstateRoute = (): void => {
+    if (window.location.pathname === ROUTES.COINS) {
+      this.renderCoinComponent();
+    }
```

Comment:
로직상 어차피 뒤의 로직이 실행되진 않겠지만, return 을 명시적으로 해줌으로써 의도를 더 명확히 나타낼 수도 있을 것 같습니다!
```suggestion
    if (window.location.pathname === ROUTES.COINS) {
      this.renderCoinComponent();
      return
    }
```

Reply:
그렇네요! 이 후 로직이 조건에 걸리지 않더라도 return을 추가해 명시적으로 하니 좀 더 명확해졌네요. 수정 완료 했습니다!